core-services/src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -73,4 +63,26 @@ public String calculatePartition(final EventType eventType, final JSONObject eve
         }
     }
 
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+            throws InvalidPartitionKeyFieldsException {
+
+        final List<String> partitionKeyFields = eventType.getPartitionKeyFields();
+        if (partitionKeyFields.isEmpty()) {
+            throw new RuntimeException("Applying " + this.getClass().getSimpleName() + " although event type " +
+                    "has no partition key fields configured.");
+        }
```

Comment:
I think it makes no sense to duplicate this check (especially as it is a RuntimeException)